From 98f39d55e77ba388f994282eaa87a69b40f39ab3 Mon Sep 17 00:00:00 2001
From: Pablo Saavedra <psaavedra@igalia.com>
Date: Mon, 7 Jun 2021 10:08:50 +0200
Subject: [PATCH] platform: Add COG_PLATFORM_HEADLESS cmake option

This CMake option allows build Cog with/without the headless platform.
The headless platform has dependency with WPEBackend-FDO not desiderable
when you are not interested on that backend for example for the case of
the DRM with the WPEBacked-RDK.

Related to: c638d133b5375ddd3c368123b39b8ab37889c921
Signed-off-by: Pablo Saavedra <psaavedra@igalia.com>
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 46ad371..9622e1b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,6 +54,7 @@ option(COG_DBUS_SYSTEM_BUS "Expose remote control interface on system bus" OFF)
 set(COG_DBUS_OWN_USER "" CACHE STRING
     "Additional user allowed to own the well-known name on the system bus")
 
+option(COG_PLATFORM_HEADLESS "Build the headless platform module" ON)
 option(COG_PLATFORM_FDO "Build the FDO platform module" ON)
 option(COG_PLATFORM_DRM "Build the DRM platform module" ON)
 option(COG_PLATFORM_X11 "Build the X11 platform module" OFF)
@@ -196,7 +197,9 @@ if (BUILD_DOCS)
     add_subdirectory(docs)
 endif ()
 
-add_subdirectory(platform/headless)
+if (COG_PLATFORM_HEADLESS)
+    add_subdirectory(platform/headless)
+endif ()
 if (COG_PLATFORM_FDO)
     add_subdirectory(platform/fdo)
 endif ()
-- 
2.27.0

